primo/utils/tests/test_clustering_utils.py

```diff
@@ -165,7 +216,7 @@ def test_perform_clustering(caplog):
     ) not in caplog.text
 
     # Capture the warning if the data has already been clustered
-    num_clusters = perform_clustering(wd)
+    num_clusters = perform_louvain_clustering(wd)
     assert num_clusters == 16
```

Comment:
Just curious about this test. Did the louvain clustering cluster the data into the same number of clusters as the agglomerative clustering method (I don't see a change to line 220)? I am curious to know why.

Reply:
Changed it to 14. The tests were not updated in this version. Please check the next commit